app/customer.py

```diff
@@ -0,0 +1,71 @@
+import datetime
+import math
+from typing import List
+from dataclasses import dataclass
+from app.car import Car
+from app.shop import Shop
+
+
+@dataclass
+class Customer:
+    name: str
+    product_cart: dict
+    location: list[int]
+    money: float | int
+    car: Car
+
+    def calculate_product_cost(self,
+                               shop: Shop) -> dict:
+        total = 0
+        purchases = {}
+        for item in self.product_cart.keys():
+            if item in shop.products.keys():
+                total += shop.products[item] * self.product_cart[item]
+                purchases[item] = round(
+                    shop.products[item] * self.product_cart[item], 2
+                )
+        return purchases
+
+    def print_receipt(self,
+                      shop: Shop,
+                      purchases: dict) -> None:
+        purchase_date = datetime.datetime.now().strftime("%d/%m/%Y %I:%M:%S")
+        print(f"Date: {purchase_date}\n"
+              f"Thanks, {self.name}, for your purchase!\n"
+              f"You have bought: ")
+
+        for key in self.product_cart.keys():

```

Comment:
Use more descriptive names and it iterates by keys by default, no need in keys()